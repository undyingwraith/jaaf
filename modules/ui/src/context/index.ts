export { AppContext, AppContextProvider } from './AppContext';
export { ThemeProvider } from './ThemeProvider';
