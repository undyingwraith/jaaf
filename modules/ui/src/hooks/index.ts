export { useOptionalService } from './useOptionalService';
export { useService } from './useService';
export { useTranslation } from './useTranslation';
