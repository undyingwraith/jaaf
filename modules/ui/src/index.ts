export * from './components';
export * from './context';
export * from './hooks';
export * from './services';
export * from './types';
export { JaafApp, type IJaafAppProps } from './JaafApp';
