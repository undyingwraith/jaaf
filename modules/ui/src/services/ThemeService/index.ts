export { type IThemeService, IThemeServiceSymbol } from './IThemeService';
export { ThemeService } from './ThemeService';
