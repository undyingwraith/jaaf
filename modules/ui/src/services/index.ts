export * from './ThemeService';
export { UiModule } from './UiModule';
