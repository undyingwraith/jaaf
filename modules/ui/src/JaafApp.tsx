import { IModule } from '@undyingwraith/jaaf-core';
import { ComponentChildren } from 'preact';
import { Router } from 'wouter-preact';
import { useHashLocation } from 'wouter-preact/use-hash-location';
import { AppContextProvider, ThemeProvider } from './context';

export interface IJaafAppProps {
	setup: IModule;
	children: ComponentChildren;
}

export function JaafApp(props: IJaafAppProps) {
	const { setup, children } = props;

	return (
		<Router hook={useHashLocation}>
			<AppContextProvider setup={setup}>
				<ThemeProvider />
				{children}
			</AppContextProvider>
		</Router>
	);
}
