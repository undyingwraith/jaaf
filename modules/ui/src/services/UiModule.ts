import { IModule } from '@undyingwraith/jaaf-core';
import { IThemeServiceSymbol, ThemeService } from './ThemeService';

export const UiModule: IModule = async (app) => {
	await app.register(ThemeService, IThemeServiceSymbol);
};
