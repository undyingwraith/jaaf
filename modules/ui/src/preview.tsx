import { render } from 'preact';
import { JaafApp } from './JaafApp';
import { Loader } from './components';

render((
	<JaafApp setup={() => { }}>
		<Loader />
	</JaafApp>
), document.getElementById('root')!);
