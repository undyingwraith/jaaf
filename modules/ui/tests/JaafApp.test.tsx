import { describe, expect, test } from 'vitest';
import { render } from '@testing-library/preact';
import { JaafApp } from '../src';
import { wait } from '../testing';
import { UiModule } from 'src/services/UiModule';

describe('JaafApp', () => {
	test('Renders correctly', async () => {
		const t = render((
			<JaafApp setup={(app) => {
				app.use(UiModule);
			}}>
				<div>TEST</div>
			</JaafApp>
		));

		await wait(500);

		expect(t.container).toMatchSnapshot();
	});
});
