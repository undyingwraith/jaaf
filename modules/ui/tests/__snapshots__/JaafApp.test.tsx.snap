// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JaafApp > Renders correctly 1`] = `
<div>
  <div
    style="--accent-color: hsl(264.70000000000005, 100%, 46.7%); --accent-color-light: hsl(264.70000000000005, 100%, 60.7%); --accent-color-dark: hsl(264.70000000000005, 100%, 32.7%); --accent-color-contrast: hsl(0, 0%, 100%); --background-color: hsl(0, 0%, 0%); --background-color-light: hsl(0, 0%, 0%); --background-color-dark: hsl(0, 0%, 0%); --background-color-contrast: hsl(0, 0%, 100%); --success-color: hsl(120, 100%, 50%); --success-color-light: hsl(120, 100%, 65%); --success-color-dark: hsl(120, 100%, 35%); --success-color-contrast: hsl(0, 0%, 0%); --info-color: hsl(240, 100%, 50%); --info-color-light: hsl(240, 100%, 65%); --info-color-dark: hsl(240, 100%, 35%); --info-color-contrast: hsl(0, 0%, 100%); --warning-color: hsl(38.80000000000001, 100%, 50%); --warning-color-light: hsl(38.80000000000001, 100%, 65%); --warning-color-dark: hsl(38.80000000000001, 100%, 35%); --warning-color-contrast: hsl(0, 0%, 0%); --error-color: hsl(0, 100%, 50%); --error-color-light: hsl(0, 100%, 65%); --error-color-dark: hsl(0, 100%, 35%); --error-color-contrast: hsl(0, 0%, 100%); background-color: var(--background-color); width: 100%; height: 100%;"
  >
    <div>
      TEST
    </div>
  </div>
</div>
`;
